src/gimelstudio/core/node/property.py

```diff
@@ -144,6 +145,29 @@ def CreateUI(self, parent, sizer):
         self.AddToFoldPanel(sizer, fold_panel, self.img)
 
 
+class ColorProp(Property):
+    """ 
+    Allows the user to select a color.
+    """
+    def __init__(self, idname, default=(255, 255, 255), visible=True):
+        Property.__init__(self, idname, default, fpb_label, expanded, visible)
+    
+    def CreateUI(self, parent, sizer):
+        fold_panel = self.CreateFoldPanel(sizer)
+        fold_panel.SetBackgroundColour(wx.Colour("#464646"))
+
+        self.colorPicker = ColorPickerButton(fold_panel,
+                                        default_value=self.default,
+                                        label=self.GetLabel())
+
+        self.AddToFoldPanel(sizer, fold_panel, self.colorpicker, spacing=10)
+        
+        self.colorPicker.Bind(EVT_NUMBERFIELD, self.WidgetEvent)
```

Comment:
Make sure to update the colorpicker variable here and on line 163 as well and use ``EVT_COLORPICKER_BUTTON`` for the event. 🙂